src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal !== this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore && herbivore.hidden === false) {
+      herbivore.health -= 50;
+
+      if (herbivore.health <= 0) {
+        herbivore.die();
```

Comment:
[CODE PERFORMANCE] - Instead of calling the `die` method which uses `filter` inside a loop (the `bite` method can be considered a loop since it may be called multiple times), it is preferred to use `filter` directly within the `bite` method after decreasing the health. This avoids the potential performance hit of multiple `filter` operations in case of many `bite` calls.